app/actions/remote/channel.test.ts

```diff
@@ -0,0 +1,762 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable max-lines */
+
+import {createIntl} from 'react-intl';
+
+import {DeepLink} from '@app/constants';
+import {SYSTEM_IDENTIFIERS} from '@app/constants/database';
+import DatabaseManager from '@database/manager';
+import NetworkManager from '@managers/network_manager';
+
+import {
+    removeMemberFromChannel,
+    fetchChannelMembersByIds,
+    updateChannelMemberSchemeRoles,
+    fetchMemberInChannel,
+    fetchChannelMemberships,
+    addMembersToChannel,
+    fetchChannelByName,
+    createChannel,
+    patchChannel,
+    leaveChannel,
+    fetchChannelCreator,
+    fetchMyChannelsForTeam,
+    fetchMyChannel,
+    joinChannel,
+    joinChannelIfNeeded,
+    markChannelAsRead,
+    unsetActiveChannelOnServer,
+    switchToChannelByName,
+    goToNPSChannel,
+    fetchMissingDirectChannelsInfo,
+    fetchDirectChannelsInfo,
+    createDirectChannel,
+    fetchChannels,
+    makeDirectChannel,
+    fetchArchivedChannels,
+    createGroupChannel,
+    fetchSharedChannels,
+    makeGroupChannel,
+    getChannelMemberCountsByGroup,
+    getChannelTimezones,
+    switchToChannelById,
+    switchToPenultimateChannel,
+    switchToLastChannel,
+    searchChannels,
+    fetchChannelById,
+    searchAllChannels,
+    updateChannelNotifyProps,
+    toggleMuteChannel,
+    archiveChannel,
+    unarchiveChannel,
+    convertChannelToPrivate,
+    handleKickFromChannel,
+    fetchGroupMessageMembersCommonTeams,
+    convertGroupMessageToPrivateChannel,
+} from './channel';
+
+import type ServerDataOperator from '@database/operator/server_data_operator';
+import type ChannelModel from '@typings/database/models/servers/channel';
+
+const serverUrl = 'baseHandler.test.com';
+let operator: ServerDataOperator;
+
+const user: UserProfile = {
+    id: 'userid',
+    username: 'username',
+    roles: '',
+} as UserProfile;
+
+let mockIsTablet: jest.Mock;
+jest.mock('@utils/helpers', () => {
+    const original = jest.requireActual('@utils/helpers');
+    mockIsTablet = jest.fn(() => false);
+    return {
+        ...original,
+        isTablet: mockIsTablet,
+    };
+});
+
+let mockGetActiveServer: jest.Mock;
+jest.mock('@queries/app/servers', () => {
+    const original = jest.requireActual('@queries/app/servers');
+    mockGetActiveServer = jest.fn(() => false);
+    return {
+        ...original,
+        getActiveServer: mockGetActiveServer,
+    };
+});
+
+const mockClient = {
```

Comment:
I wonder if it would be better to use nock as in https://github.com/mattermost/mattermost-mobile/blob/main/test/test_helper.ts#L527

Reply:
I was following the example set here: https://github.com/mattermost/mattermost-mobile/blob/main/app/products/calls/actions/calls.test.ts#L41

0/5 whether to use nock instead. Since it's a unit test I think it's fine to mock at a shallower level like I've done. Unless there's a strong opinion to change it to nock then I'll leave it as is

Reply:
Is ok, approved already so no strong opinion